api/v1/mongodbcommunity_types.go

```diff
@@ -736,29 +761,96 @@ func (m *MongoDBCommunity) AutomationConfigArbitersThisReconciliation() int {
 	})
 }
 
+// GetOptionsString return a string format of the connection string
+// options that can be appended directly to the connection string.
+//
+// Only takes into account options for the resource and not any user.
+func (m *MongoDBCommunity) GetOptionsString() string {
+	generalOptionsMap := m.Spec.AdditionalConnectionStringConfig.Object
+	optionValues := make([]string, len(generalOptionsMap))
+	i := 0
+
+	for key, value := range generalOptionsMap {
+		if key != "replicaSet" && key != "ssl" && key != "tls" {
+			optionValues[i] = fmt.Sprintf("%s=%v", key, value)
+			i += 1
+		}
+	}
+
+	optionValues = optionValues[:i]
+
+	optionsString := ""
+	if i > 0 {
+		optionsString = "&" + strings.Join(optionValues, "&")
+	}
+	return optionsString
+}
+
+// GetUserOptionsString return a string format of the connection string
+// options that can be appended directly to the connection string.
+//
+// Takes into account both user options and resource options.
+// User options will override any existing options in the resource.
+func (m *MongoDBCommunity) GetUserOptionsString(user scram.User) string {
+	generalOptionsMap := m.Spec.AdditionalConnectionStringConfig.Object
+	userOptionsMap := user.ConnectionStringOptions
+	optionValues := make([]string, len(generalOptionsMap)+len(userOptionsMap))
+	i := 0
+	for key, value := range userOptionsMap {
+		if key != "replicaSet" && key != "ssl" && key != "tls" {
```

Comment:
It would be nice to extract the protected options in a constant for easier maintenance.

Reply:
Done! However, Go doesn't allow constant maps or any other reference type for that matter. So I used a variable map, since having multiple string constants wouldn't be easily maintainable.